docs/core-migration-guide.md

```diff
@@ -42,7 +42,7 @@ license: |
 
 - Since Spark 4.0, Spark uses the external shuffle service for deleting shuffle blocks for deallocated executors when the shuffle is no longer needed. To restore the legacy behavior, you can set `spark.shuffle.service.removeShuffle` to `false`.
 
-- Since Spark 4.0, the default log4j output has shifted from plain text to JSON lines to enhance analyzability. To revert to plain text output, you can either set `spark.log.structuredLogging.enabled` to `false`, or use a custom log4j configuration.
```

Comment:
Note: the flag `spark.log.structuredLogging.enabled` is not taking effect as expected. I created https://issues.apache.org/jira/browse/SPARK-48126 as a follow-up. Let's not mention it in the Spark 4.0 preview release.